python/cudf/cudf/io/parquet.py

```diff
@@ -64,11 +64,38 @@ def read_parquet(
 
 
 @ioutils.doc_to_parquet()
-def to_parquet(df, path, *args, **kwargs):
+def to_parquet(
+    df,
+    path,
+    engine="cudf",
+    compression="snappy",
+    index=None,
+    partition_cols=None,
+    statistics="ROWGROUP",
+    *args,
+    **kwargs,
+):
     """{docstring}"""
-    warnings.warn(
-        "Using CPU via PyArrow to write Parquet dataset, this will "
-        "be GPU accelerated in the future"
-    )
-    pa_table = df.to_arrow()
-    pq.write_to_dataset(pa_table, path, *args, **kwargs)
+
+    if engine == "cudf":
+
+        if index is not None:
+            ValueError(
+                "'index' is currently not supported by the gpu "
+                + "accelerated parquet writer"
+            )
+
+        if partition_cols is not None:
+            ValueError(
+                "'partition_cols' is currently not supported by the "
+                + "gpu accelerated parquet writer"
+            )
+
+        return libcudf.parquet.write_parquet(
+            df, path, compression=compression, statistics=statistics
+        )
+    else:
+        pa_table = df.to_arrow(preserve_index=index)
```

Comment:
`preserve_index` argument defaults to `True` and I don't believe properly handles `None`. May want to check what the default behavior of Pandas is here and set `index` accordingly before passing to here.